tests/s3_data_plane_tests.c

```diff
@@ -398,6 +400,90 @@ static int s_test_s3_get_object_multiple(struct aws_allocator *allocator, void *
     return 0;
 }
 
+int test_s3_get_object_sse_helper(struct aws_allocator *allocator, enum AWS_S3_TESTER_SSE_TYPE sse_type) {
+
+    const struct aws_byte_cursor test_object_path =
```

Comment:
Would it be possible to do something like you did for aws_s3_tester_send_put_object_meta_request but for aws_s3_tester_send_get_object_meta_request?  Feels like a big copy paste for just the if at the top. :)